tools/org.eclipse.n4js.hlc.base/src/org/eclipse/n4js/hlc/base/N4jscBase.java

```diff
@@ -916,11 +918,15 @@ private void initInjection(Properties properties) {
 		N4mfPackage.eINSTANCE.getNsURI();
 		XMLTypePackage.eINSTANCE.getNsURI();
 
-		final Module cliTesterModule = override(new TesterModule())
-				.with((Module) binder -> binder.bind(TestTreeTransformer.class).to(CliTestTreeTransformer.class));
-
-		final Module module = Modules.combine(new N4JSRuntimeModule(), cliTesterModule,
-				new N4JSHeadlessGeneratorModule(properties));
+		final Module module = override(
+				Modules.combine(new N4JSRuntimeModule(),
+						new TesterModule(),
+						new N4JSHeadlessGeneratorModule(properties)))
+								.with((Module) binder -> {
+									binder.bind(TestTreeTransformer.class).to(CliTestTreeTransformer.class);
+									binder.bind(IHeadlessLogger.class)
+											.toInstance(new ConfigurableHeadlessLogger(verbose, debug));
+								});
```

Comment:
I find it difficult to understand what happens here, because there are so many nested expressions.